prophesee_ros_driver/src/prophesee_ros_publisher.cpp

```diff
@@ -185,9 +274,9 @@ void PropheseeWrapperPublisher::publishGrayLevels() {
 
             // Define the message for the gray level frame
             cv_bridge::CvImage gl_frame_msg;
-            gl_frame_msg.header.stamp = ros::Time::now();
-            gl_frame_msg.header.frame_id = "PropheseeCamera_optical_frame";
-            gl_frame_msg.encoding = sensor_msgs::image_encodings::TYPE_8UC1;
+            gl_frame_msg.header.stamp = last_timestamp_;
```

Comment:
You are right. However with IMU timestamps the IMU-Frame timeshift is more accurate (timeshift is 6ms tested with kalibr).  IMU-Frame timeshift is 10x higher using ROS "independent" timestamp (60ms).
So my solution here is, keep IMU topic True as default since all cameras have IMU. Therefore, to use the hardware IMU timestamp as the Frame timestamp. Alternatively, I could use the last CD event timestamp in case you prefer. I already tested this solution and the timeshift is similar approx. 6ms.

Reply:
Hi Javier, better the last CD event.
I think it is more logic and it can be used also when the IMU is not available.

Reply:
agree with Davide.
As not all of our camera products has IMU

Reply:
@jhidalgocarrio could you please address this issue ? as not all our cameras have IMU

Reply:
ok , working on that

Reply:
thanks !